var/spack/repos/builtin/packages/jsoncpp/package.py

```diff
@@ -65,3 +65,8 @@ def cmake_args(self):
         else:
             args.append('-DJSONCPP_WITH_TESTS=OFF')
         return args
+
+    def check(self):
+        # TODO: known test failures
+        # see https://github.com/spack/spack/issues/24773
+        pass
```

Comment:
Hmm, the issue says that the tests are failing - not why they are failing. If the failure is legitimate and not an issue known to upstream, we shouldn't skip this check since it's not a false positive.

Reply:
I had a slack discussion with @adamjstewart (I think) about this. The issue is that there are [*a lot*](https://github.com/spack/spack/issues?q=is%3Aissue+is%3Aopen++%22Test+Issue%3A%22++author%3Avvolkl) of packages with broken tests, which renders `--test=all` fairly unusable. 

As also discussed on slack, Spack needs a more fine-grained control over running tests than `--test=all` and `--test=root`, and also an option to ignore test failures. I'm working on both, but until then I think it's better to skip one or two tests in order to allow people to actually use `spack install --test=all`, it's a terrible experience to have to fix tests on dependencies and restart builds over and over again. The issue hopefully means that this is not forgotten. I can maybe also add a warning printout?

Reply:
My philosophy has been to report all test failures upstream, and if they say to ignore it then I'll skip the tests. Otherwise, I may just add a comment to the package saying that the tests may fail. Unless you develop the software yourself, it's almost impossible to tell whether the test failures are critical or ignorable.

Reply:
That sounds good. I think I'll try to add the possibility to more selectively run tests, then they don't need to be disabled. In this case I took a look at the upstream bug tracker and it's likely we're seeing this issue https://github.com/open-source-parsers/jsoncpp/issues/1136
 - tests look like they have been recently fixed for meson, but not cmake: https://github.com/open-source-parsers/jsoncpp/issues/1102#issuecomment-619163645 

I'll update the issue. By the way, is there a decorator or something to disable tests for some specific version with known test failures?

Reply:
You could add `@when('@:1.2.3')` to `def check` to disable it for a range of versions.

Reply:
@vvolkl